central/delegatedregistryconfig/delegator/delegator.go

```diff
@@ -60,7 +60,11 @@ func (d *delegatorImpl) GetDelegateClusterID(ctx context.Context, imgName *stora
 		return "", false, nil
 	}
 
-	if err := d.validateCluster(clusterID); err != nil {
+	if clusterID == "" {
+		return "", true, errors.New("no ad-hoc cluster specified in delegated registry config")
```

Comment:
Got two points on this one:
- we try to use our sentinel errors (a.k.a roxerrors) where possible. Not sure any existing one fits here, but worth taking a look (maybe invalid args?).
- first time reading this error in the PR description, it wasn't really clear what to do with this error. Should we hint that a cluster ID is required?

Reply:
> first time reading this error in the PR description, it wasn't really clear what to do with this error. Should we hint that a cluster ID is required?

That error is an existing error, just moved due to the expanded usage of the `ValidateCluster`.  Since only ad-hoc requests would encounter this error it would be shown inline via the API response and with roxctl, believe in those cases its straitforward:

```sh
$ roxctl image scan --image=<something>
ERROR:	Scanning image failed: could not scan image: "nginx:1.14.0": rpc error: code = Internal desc = no ad-hoc cluster specified in delegated registry config. Retrying after 3 seconds...
```

I've updated the error message to `no ad-hoc cluster ID specified in the delegated scanning config`, although no longer 'accurate' per code - the new UI and docs refer to the feature as `delegated scanning`.